rskj-core/src/test/java/org/ethereum/crypto/signature/ECDSASignatureTest.java

```diff
@@ -0,0 +1,163 @@
+    }
+
+    @Test
+    public void testEquals() {
+        ECDSASignature expected = new ECDSASignature(BigInteger.ONE, BigInteger.ONE, (byte) 27);
+
+        //same instance
+        assertTrue(expected.equals(expected));
+
+        //same values
+        assertTrue(expected.equals(new ECDSASignature(expected.getR(), expected.getS(), expected.getV())));
+
+        //same values - but diff v
+        assertTrue(expected.equals(new ECDSASignature(expected.getR(), expected.getS(), (byte) 0)));
+
+        // null
+        assertFalse(expected.equals(null));
+
+        //dif classes
+        assertFalse(expected.equals(BigInteger.ZERO));
+
+        //diff r
+        assertFalse(new ECDSASignature(BigInteger.ONE, BigInteger.ONE, (byte) 27).equals(new ECDSASignature(BigInteger.TEN, BigInteger.ONE, (byte) 27)));
+
+        //diff s
+        assertFalse(new ECDSASignature(BigInteger.ONE, BigInteger.ONE, (byte) 27).equals(new ECDSASignature(BigInteger.ONE, BigInteger.TEN, (byte) 27)));
+
+    }
+
+    @Test
+    public void testValidateComponents_SignedMsg() {
+        ECKey key = new ECKey();
+        byte[] hash = HashUtil.keccak256(exampleMessage.getBytes());
+        ECDSASignature signature = ECDSASignature.fromSignature(key.sign(hash));
+        assertTrue(signature.validateComponents());
+    }
+
+    @Test
+    public void fromComponentsWithRecoveryCalculation() {
+        ECKey key = new ECKey();
+        byte[] hash = HashUtil.randomHash();
+        ECDSASignature signature = ECDSASignature.fromSignature(key.sign(hash));
+
+        // With uncompressed public key
+        ECDSASignature signatureWithCalculatedV = fromComponentsWithRecoveryCalculation(
+                signature.getR().toByteArray(),
+                signature.getS().toByteArray(),
+                hash,
+                key.getPubKey(false)
+        );
+
+        Assert.assertEquals(signature.getR(), signatureWithCalculatedV.getR());
+        Assert.assertEquals(signature.getS(), signatureWithCalculatedV.getS());
+        Assert.assertEquals(signature.getV(), signatureWithCalculatedV.getV());
+
+        // With compressed public key
+        signatureWithCalculatedV = fromComponentsWithRecoveryCalculation(
+                signature.getR().toByteArray(),
+                signature.getS().toByteArray(),
+                hash,
+                key.getPubKey(true)
+        );
+
+        Assert.assertEquals(signature.getR(), signatureWithCalculatedV.getR());
+        Assert.assertEquals(signature.getS(), signatureWithCalculatedV.getS());
+        Assert.assertEquals(signature.getV(), signatureWithCalculatedV.getV());
+    }
+
+
+    /**
+     * @param r    -
+     * @param s    -
+     * @param hash - the hash used to compute this signature
+     * @param pub  - public key bytes, used to calculate the recovery byte 'v'
+     * @return -
+     */
+    public static ECDSASignature fromComponentsWithRecoveryCalculation(byte[] r, byte[] s, byte[] hash, byte[] pub) {
```

Comment:
remove these two methods and point to the actual methods in ECDSASignature

Reply:
Done